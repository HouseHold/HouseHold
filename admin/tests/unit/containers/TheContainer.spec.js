import CoreuiVue from '@coreui/vue'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router'
import TheContainer from '@/containers/TheContainer'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(CoreuiVue)

describe('TheContainer.vue', () => {
  it('has a name', () => {
    expect(TheContainer.name).toBe('TheContainer')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(TheContainer, {
      localVue,
        router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(TheContainer, {
      localVue,
      router
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })
})
