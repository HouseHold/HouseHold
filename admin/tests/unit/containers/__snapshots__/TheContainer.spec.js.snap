// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheContainer.vue renders correctly 1`] = `
<div
  class="c-app"
>
  <thesidebar-stub />
   
  <div
    class="c-wrapper"
  >
    <theheader-stub />
     
    <div
      class="c-body"
    >
      <main
        class="c-main"
      >
        <ccontainer-stub
          fluid="true"
          tag="div"
        >
          <router-view-stub
            name="fade"
          />
        </ccontainer-stub>
      </main>
    </div>
     
    <thefooter-stub />
  </div>
</div>
`;
